vault/identity_store_oidc_provider_util.go

```diff
@@ -0,0 +1,49 @@
+package vault
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+func isValidRedirectURI(uri string, validUris []string) bool {
```

Comment:
We have a function that provides this validation over in our OIDC auth method (see [path_oidc.go#L514-L543](https://github.com/hashicorp/vault-plugin-auth-jwt/blob/master/path_oidc.go#L514-L543)). I'd prefer to reuse that function by copying it over into this new file instead of writing something new.